slsim/Halos/halos_lens_base.py

```diff
@@ -496,9 +496,6 @@ def _build_lens_data(self, halos, mass_correction, z1, z2):
         # If this above code need to be changed, notice the change in the following code
         # including the lens_cosmo_dict one since it assume halos is in front of mass sheet
         if not combined_redshift_list.size:
-            warnings.warn(
```

Comment:
I think the main warning that was printed thousands of time during testing is this one. The other ones you might leave, but it's up to you - thank you @trivialTZ !